components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2163,39 +2106,30 @@ void KeyringService::GetFilecoinSelectedAccount(
   std::move(callback).Run(GetFilecoinSelectedAccount(net));
 }
 
-void KeyringService::SetSelectedAccount(const std::string& address,
-                                        mojom::CoinType coin,
+void KeyringService::SetSelectedAccount(mojom::CoinType coin,
+                                        const std::string& keyring_id,
+                                        const std::string& address,
                                         SetSelectedAccountCallback callback) {
-  auto keyring_id = GetKeyringId(coin, address);
-
   std::vector<mojom::AccountInfoPtr> infos =
       GetAccountInfosForKeyring(keyring_id);
 
-  // Check for matching default and imported account
   for (const mojom::AccountInfoPtr& info : infos) {
     if (base::EqualsCaseInsensitiveASCII(info->address, address)) {
-      SetSelectedAccountForCoin(coin, address);
+      SetSelectedAccountForCoin((*info));
       std::move(callback).Run(true);
       return;
     }
   }
 
-  auto hardware_account_info_ptrs = GetHardwareAccountsSync(keyring_id);
-  for (const mojom::AccountInfoPtr& info : hardware_account_info_ptrs) {
-    if (base::EqualsCaseInsensitiveASCII(info->address, address)) {
-      SetSelectedAccountForCoin(coin, address);
-      std::move(callback).Run(true);
-      return;
-    }
-  }
```

Comment:
GetAccountInfosForKeyring also returns hardware accounts, so this block was no-op